src/sources/kubernetes_logs/pod_metadata_annotator.rs

```diff
@@ -130,6 +136,24 @@ fn annotate_from_pod_spec(log: &mut LogEvent, fields_spec: &FieldsSpec, pod_spec
     }
 }
 
+fn annotate_from_pod_status(log: &mut LogEvent, fields_spec: &FieldsSpec, pod_status: &PodStatus) {
+    k8s_if_le_1_16! {
+        for (ref key, ref val) in [(&fields_spec.pod_ips, &pod_status.pod_ip)].iter() {
+            if let Some(val) = val {
+                log.insert(key, val.to_owned());
+            }
+        }
+    }
+    k8s_if_ge_1_15! {
```

Comment:
This is a compile-type check, and we don't allow specializing the `kube-openapi` features. We also don't need it.

I'd suggest that we simply set *both* `pod_ips` and `pod_ip` in the event. If both the `pod_status.pod_ip` and `pod_status.pod_ips` are present - we'll have some duplication in the event, but it's fine since it's up to the Kubernetes project decide how to deal with this duplication, and I'd say we should avoid intervening with custom logic here - and just pass values as is. It's a wise choice to keep the implementation direct, and the behaviour easy to understand for the users.

So, I suggest we just have something like this:

```rust
        for (ref key, ref val) in [(&fields_spec.pod_ip, &pod_status.pod_ip)].iter() {
            if let Some(val) = val {
                log.insert(key, val.to_owned());
            }
        }
        for (ref key, ref val) in [(&fields_spec.pod_ips, &pod_status.pod_ips)].iter() {
            if let Some(val) = val {
                let inner: Vec<String> = val.iter().filter_map(|v| v.ip.clone()).collect::<Vec<String>>();
                // assign an array here
            }
        }
```

In both cases, we just pass through the values without any kind of custom logic to derive a single value. KISS way.